File: scripts/component_template/component_name.ts
import {ChangeDetectionStrategy, Component, Input} from '@angular/core';
import {
  Key,
  Type,
  UserEvent,
} from 'mesop/mesop/protos/ui_jspb_proto_pb/mesop/protos/ui_pb';
import {ComponentNameType} from 'mesop/mesop/components/component_name/component_name_jspb_proto_pb/mesop/components/component_name/component_name_pb';
import {Channel} from '../../web/src/services/channel';

@Component({
  selector: 'mesop-{component-name}',
  templateUrl: '{component_name}.ng.html',
  standalone: true,
})
export class ComponentNameComponent {
  @Input({required: true}) type!: Type;
  @Input() key!: Key;
  private _config!: ComponentNameType;

  ngOnChanges() {
    this._config = ComponentNameType.deserializeBinary(
      this.type.getValue() as unknown as Uint8Array,
    );
  }

  config(): ComponentNameType {
    return this._config;
  }
}

File: scripts/component_template/component_name.py
import mesop.components.component_name.component_name_pb2 as component_name_pb
from mesop.component_helpers import insert_component, register_native_component


@register_native_component
def component_name(
  *,
  label: str,
  key: str | None = None,
):
  """
  This function creates a component_name.

  Args:
      label: The text to be displayed
  """
  insert_component(
    key=key,
    type_name="component_name",
    proto=component_name_pb.ComponentNameType(
      label=label,
    ),
  )
